src/charm.py

```diff
@@ -227,6 +245,64 @@ def _on_database_storage_detaching(self, _) -> None:
         # Inform other hooks of current status
         self._peers.data[self.unit]["unit-status"] = "removing"
 
+    def _on_update_status(self, _) -> None:
+        """Handle update status.
+
+        Takes care of workload health checks and.
+        """
+        if not self.cluster_initialized or not self.unit_peer_data.get("member-role"):
+            # health checks only after cluster and member are initialised
+            return
+
+        # retrieve and persist state for every unit
+        try:
+            state, role = self._mysql.get_member_state()
+            self.unit_peer_data["member-role"] = role
+            self.unit_peer_data["member-state"] = state
+        except MySQLGetMemberStateError:
+            if self.unit_peer_data.get("member-state") == "waiting":
+                # avoid changing status while in initialisation
+                return
+            role = self.unit_peer_data["member-role"] = "unknown"
+            state = self.unit_peer_data["member-state"] = "unreachable"
+        logger.info(f"Unit workload member-state is {state} with member-role {role}")
+
+        # set unit status based on member-{state,role}
+        self.unit.status = (
+            ActiveStatus("R/W" if role == "primary" else "")
```

Comment:
If you're showing whether the nodes are RW in the juju status, is it worth setting all the secondaries to "R"? Otherwise, this UI implies that you can only read from the leader.

Reply:
:thinking:  I may replicate what the legacy charm does, i.e.: `Unit is ready: Mode: RO` (or RW)